nativelib/src/main/scala/scala/scalanative/cre2/cre2.scala

```diff
@@ -0,0 +1,138 @@
+package java.util.regex
+
+import scala.scalanative._
+import native._
+import native.stdlib._
+import runtime.GC
+
+import java.nio.charset.Charset
+
+// Syntax guide: https://github.com/google/re2/wiki/Syntax
+// re2 c wrapper: https://github.com/marcomaggi/cre2
+@link("re2")
+@extern
+private[regex] object cre2 {
+  import cre2h._
+
+  def cre2_new(
+      pattern: CString,
+      pattern_len: CInt,
+      options: Ptr[cre2_options_t]
+  ): Ptr[cre2_regexp_t] = extern
+
+  def cre2_delete(
+      regex: Ptr[cre2_regexp_t]
+  ): Unit = extern
+
+  def cre2_quote_meta(
+      quoted: cre2_string_t,
+      original: cre2_string_t
+  ): CInt = extern
+
+  def cre2_num_capturing_groups(
+      regex: Ptr[cre2_regexp_t]
+  ): CInt = extern
+
+  def cre2_find_named_capturing_groups(
+      regex: Ptr[cre2_regexp_t],
+      name: CString
+  ): CInt = extern
+
+  def cre2_match(
+      regex: Ptr[cre2_regexp_t],
+      text: CString,
+      textlen: CInt,
+      startpos: CInt,
+      endpos: CInt,
+      anchor: cre2_anchor_t,
+      matches: cre2_string_t,
+      nMatches: CInt
+  ): CInt = extern
+
+  def cre2_replace_re(
+      regex: Ptr[cre2_regexp_t],
+      text_and_target: cre2_string_t,
+      rewrite: cre2_string_t
+  ): CInt = extern
+
+  def cre2_global_replace_re(
+      regex: Ptr[cre2_regexp_t],
+      text_and_target: cre2_string_t,
+      rewrite: cre2_string_t
+  ): CInt = extern
+
+  // TODO: #598 (remove this)
+  def cre2_distance(a: CString, b: CString): CInt = extern
```

Comment:
The issue is #598. This is an indication that when #598 is implemented this code can be removed.